### ISSUE TITLE
String damp indications for harp

### ISSUE THREAD

**Issue by rettinghaus** [music-encoding/music-encoding#845]
In scores hand damping symbols can pop up everywhere, they are rarely connected to notes. 

![dampall](https://user-images.githubusercontent.com/7693447/127176034-1dc74efb-f857-4b72-8caf-78d86d81d01d.png)
![dampsingle_string](https://user-images.githubusercontent.com/7693447/127176036-72cddd12-a66c-4a30-b5eb-c378e510db6f.png)
![Image](https://user-images.githubusercontent.com/7693447/127177549-ec12694f-d77e-494e-a783-233d8494695c.png)

Yet in MEI it is only possible to encode them as `artic`.
In modern notation they are sometimes to something that looks like a glissando to muffle a particular group of notes. 
(See [The Sacred Harp Notation Guide](https://www.harpspectrum.org/harpworks/composing_for_harp/images/sacred_harp_notation.gif) and section _Hand damping_ in Gould pp. 364-366.)

I propose to remove the values `damp` and `dampall` from `data.ARTICULATION` and have another control element for that purpose. Maybe it could/should be combined with what is currently `harpPedal` (see #643).

**Comment by ahankinson** [comment]
`artic` can occur directly in `layer`, and has `@plist` so you can attach the 'active participant' IDs.

It can also take various `@tstamp` values. Additionally, `@artic` could be modified to take `@startid`?

Just proposing alternate solutions...

**Comment by rettinghaus** [comment]
Good point. But is harp damping an articulation at all? Or why are there special elements for piano pedal and laissez vibrer, which are closer related to harp damping than a marcato.

**Comment by ahankinson** [comment]
🤷

**Comment by ahankinson** [comment]
If we do choose to do a new element, I propose it *not* be called `<damp>`. Otherwise the brass might mistake it as a direction to empty their spit valves. 😀

**Comment by pe-ro** [comment]
The statement that "Yet in MEI it is only possible to encode them as artic." isn't true.  In MEI "secondary" entities, such as articulations, can either be associated with a "primary" entity, either via attributes or child elements of the primary entity, or associated with a staff, through the use of a so-called "control event".  When there is no specific control event, as is the case with damping symbols, one can/should use the generic control event `<dir>` with appropriate content -- in this case probably `<symbol>`.  Position and all kinds of other stuff can be recorded about the directive using the attributes of `<dir>`.

Some entities, such as `<artic>`, have been elevated from attributes and/or event children to control event status.  One justification for this, among others, is that the entity requires unique attributes of its own.  Obviously, every attribute or child could be elevated in this way, but doing so would cause a tremendous explosion in the number of new control events.  I admit haven't thought about it a lot, and I'd be happy to be proven wrong, but I don't think elevating `@artic` values of 'damp' and 'dampall' to `<damp>` (sorry, Andrew) is justified.

**Comment by pe-ro** [comment]
If one thinks that damping symbols aren't articulations, then using `<artic>` to encode them would be "tag abuse".

**Comment by lpugin** [comment]
In my opinion they do look like control events. So maybe another idea would be to have `damp` to qualify a `<dir>` as suggested here https://github.com/music-encoding/music-encoding/issues/712 ?

**Comment by pe-ro** [comment]
@lpugin, that's fine, but adding `@func` will lead to confusion between it and `@type` even if `@func` uses a closed list of values.  I prefer a semi-closed value list for `@type`.

**Comment by lpugin** [comment]
I don't. I prefer to separate them and leave `@type` totally free for project specific uses.

**Comment by pe-ro** [comment]
That's fine until the values in `@type` and `@func` are the same.  For example, if I use "damp" as the value of `@type`, Verovio will ignore my value until I put it in `@func`.  But doing so will a) cause me to repeat data, and/or force me to change my project-specific needs.

On the other hand, if there's a single place to put project-specific and Verovio-specific info (which is what `@func` is, I think) , then Verovio (or another app) can determine its behavior based on whether an approved value has been used or not *and* encoder confusion has been avoided completely.  If an approved value has not been used, Verovio can gleefully ignore it.

**Comment by lpugin** [comment]
My point is that if we start (again) to fill `@type` with pre-defined values, then `@type` is not completely free for project specific needs. In other words, documenting its use will be more difficult, and that will introduce confusion. Your argument of putting `damp` in `@type` and introducing then duplication with `@func` or the need for the data to be changed could be used for any attribute (e.g., `<accid func="edit" type="edit">`)

Nothing is meant to be Verovio-specific, even though I understand that anything coming from me will look like it 🤣. Verovio can without problem handle values in `@type`, with no list, an open list, or a close list. It is only an idea to improve the semantic in MEI, and I am happy to hear that it is a bad one.

**Comment by rettinghaus** [comment]
The problem by using `dir` is that this would be a visual only approach, which isn't very machine readable. What about e.g. MIDI information? And you could only encode the simple sign, but would not be able to add information about the affected pitch range. 
Arguing that basically everything can be encoded with `dir`, I wonder then why we do have specialized elements at all? Why is there a dedicated element for piano pedaling, and not a similar element for harp? Only because there is much more piano music than harp music available? When we encounter commonly used specialized notation, there has to be a way to address this specifically.

**Comment by pe-ro** [comment]
Of course, there has to be a way to encode specialized directives and I don't oppose creating special elements when necessary.  The discussion at the moment is about whether it's necessary to create a new `<damp>` element.  What specific characteristics / attributes would you like `<damp>` to have that `<dir>` doesn't already supply?  MIDI is a bit of a canard, so it would be best to avoid it right now, but affected pitch(es) might be worth recording.  Why isn't `@plist` adequate for this?

**Comment by kepper** [comment]
This whole discussion reminds me about the argument I've given [elsewhere](https://github.com/music-encoding/music-encoding/pull/846#issuecomment-889022967): Sometimes, music notation simply doesn't want to follow our criteria for what is a control event and what is a note modifier. I'm fully aware that opening the door or blurring that line will cause madness, but maybe this is just the madness of the thing we're trying to encode? 
If we'd agree that something along these lines is the underlying problem, then we should not rush into a solution here, but instead take some time to think this through as much as possible. We have quite some such topics in the backlog that need intense discussions and final decisions, but we're usually lacking the time to cover them as carefully as necessary. Maybe it's time for a dedicated modelling workshop, but I can't see how this should be accomplished in digital format…

**Comment by ahankinson** [comment]
I don't really see a difference between a control event and a note modifier. As mentioned, `<artic>` can be contained by both layer and note. If you need to express it with standoff notation, then that's OK. If you want to apply it to a specific note, then that's OK too. 

I would second @lpugin's wish not to use `@type`. I too remember saying "Oh, just use `@type` for ... ". I also seem to remember pretty strong arguments to not propagate that once they were cleaned up, and the introduction of `@nnnntype` attributes (like `@notationtype`, etc.) so that we could be specific about what is meant by the value carried in type.

The suggestion about adapting `harpPedal` is interesting, but with my relatively poor knowledge of harp playing techniques, is this OK? Are there other ways of damping harps other than a pedal? Can you dampen the strings with your hands? Does that have a different indication?

**Comment by pe-ro** [comment]
Harp pedals are used to select the pitches of the strings.  They have nothing to do with damping, which is done entirely with the hands.

**Comment by rettinghaus** [comment]
Well I could imagine a composer wanting the player to dampen the strings with the feet. 
But what I was thinking of is to have an element for harp techniques, so rename `harPedal` to simply `harp` (or something similar) and use that for all things the harpist has to do except from plucking the strings (which of course would still be indicated by the notes).

**Comment by pe-ro** [comment]
Sorry, but that's going in the wrong direction.  Pedal settings can't be mixed with general technique indications because harp pedaling is specific, e.g., has specific function and data requirements.  `<harpPedal>` sets which pitches are actually available and is used to render a specific diagram showing the pedal settings.  When used for damping, a generic `<harp>` element would be overburdened with the attributes necessary for pedaling --
```
<harp c="f" d="f" e="f" f="f" g="f" a="f" b="f"> [damp sign] </harp>
```
This will only lead to confusion.

**Comment by pe-ro** [comment]
Given the decision made in Berlin this weekend about adding `<repeatMark>` , I think it's ok to move forward with creating a `<damp>` element based on `<dir>`.  I think it can use the same content model and attributes as `<dir>`.
